fplc_controller.py

```
@@ -0,0 +1,381 @@
+        """
+        Main control loop that runs periodically.
+        Reads sensor data, applies MPC-like logic, and sends commands.
+        """
+        if not self._is_running or self._is_paused:
+            return
+
+        current_time_in_step = (time.time() - self._step_start_time) / 60.0 # in minutes
+        current_run_time = (time.time() - self._run_start_time) / 60.0 # in minutes
+
+        # 1. Read current sensor data
+        current_uv = self.uv_detector.get_reading()
```

Ruff (0.12.2)

Local variable `current_uv` is assigned to but never used

Remove assignment to unused variable `current_uv`

(F841)